packages/itmat-cores/src/trpcCore/driveCore.ts

```diff
@@ -0,0 +1,832 @@
+import { DBType } from '../database/database';
+import { enumUserTypes, enumFileTypes, enumFileCategories, IDrive, enumDriveNodeTypes, IFile, IDrivePermission, FileUpload, CoreError, enumCoreErrors, IUserWithoutToken, enumConfigType, defaultSettings, IUserConfig } from '@itmat-broker/itmat-types';
+import { v4 as uuid } from 'uuid';
+import { TRPCFileCore } from './fileCore';
+import { UpdateFilter } from 'mongodb';
+import { ObjectStore } from '@itmat-broker/itmat-commons';
+import { makeGenericReponse } from '../utils';
+
+export class TRPCDriveCore {
+    db: DBType;
+    fileCore: TRPCFileCore;
+    objectStore: ObjectStore;
+    constructor(db: DBType, fileCore: TRPCFileCore, ObjectStore: ObjectStore) {
+        this.db = db;
+        this.fileCore = fileCore;
+        this.objectStore = ObjectStore;
+    }
+    /**
+     * Create a drive folder.
+     *
+     * @param requester - The requester.
+     * @param folderName - The name of the folder.
+     * @param parentId - The id of the parent. Null for default root node.
+     * @param protected - Whether this folder is protected. Protected folder could not be deleted.
+     * @param description - The description of the folder.
+     *
+     * @return IDrive
+     */
+    public async createDriveFolder(requester: IUserWithoutToken | undefined, folderName: string, parentId: string | null, restricted: boolean, description?: string) {
```

Comment:
We just need to understand the typing information problem here. As discussed, I'm adding a comment to have a reference 😉